### PR TITLE
DM-6141: improve the mouse control with the drawing layers

### PR DESCRIPTION
**Pull request by robyww**
- mouse selection should now work the way it is expected to
 - includes bug fixes with distance tool and select area editing
 - includes layer fixes, the most active should be on top

### CODE REVIEW

File: src/firefly/js/drawingLayers/SelectArea.js [hunk 1]
```diff
@@ -132,7 +168,7 @@ function moveMouse(drawLayer,action) {
     var mode= getMode(plot);
     if (plot && mode==='edit') {
         var cc= CsysConverter.make(plot);
-        var ptAry= getPtAry(plot);
+        var ptAry= getPtAryFromPlot(plot);
```
Comment (cwang2016) [inline src/firefly/js/drawingLayers/SelectArea.js]: should here also checks ptAry (i.e. if (!ptAry) ... like line 205) ?

**Comment by cwang2016** [comment]
looks good to me.

File: src/firefly/js/drawingLayers/SelectArea.js (hunk 1 above)
Comment (robyww) [inline src/firefly/js/drawingLayers/SelectArea.js]: I think so.  I will fix it.

File: src/firefly/js/drawingLayers/SelectArea.js [hunk 2]
```diff
@@ -69,31 +75,60 @@ function creator() {
...
 }
 
 function onDetach(drawLayer,action) {
     var {plotIdAry}= action.payload;
     plotIdAry.forEach( (plotId) => dispatchAttributeChange(plotId,false,PlotAttribute.SELECTION,null,true));
 }
 
+function getCursor(plotView, screenPt) {
+    const plot= primePlot(plotView);
+    var cc= CsysConverter.make(plot);
```
Comment (lznakano) [inline src/firefly/js/drawingLayers/SelectArea.js]: Should add this line here? 
if (!cc) return;

File: src/firefly/js/drawingLayers/SelectArea.js [hunk 3]
```diff
@@ -132,7 +168,7 @@ function moveMouse(drawLayer,action) {
     var mode= getMode(plot);
     if (plot && mode==='edit') {
         var cc= CsysConverter.make(plot);
```
Comment (lznakano) [inline src/firefly/js/drawingLayers/SelectArea.js]: Should check cc as if (!cc) return;?

File: src/firefly/js/drawingLayers/SelectArea.js [hunk 4]
```diff
@@ -188,20 +224,32 @@ function start(drawLayer,action) {
...
     if (!sel) return null;
-    var cc= CsysConverter.make(plot);
+    return getPtAry(plot,sel.pt0,sel.pt1);
+}
+
+function getPtAry(plot,pt0,pt1) {
     var ptAry=[];
-    ptAry[0]= cc.getScreenCoords(sel.pt0);
-    ptAry[2]= cc.getScreenCoords(sel.pt1);
+    var cc= CsysConverter.make(plot);
```
Comment (lznakano) [inline src/firefly/js/drawingLayers/SelectArea.js]: Should cc be checked here as well?

File: src/firefly/js/drawingLayers/SelectArea.js (hunk 3 above)
Comment (robyww) [inline src/firefly/js/drawingLayers/SelectArea.js]: I am already checking plot.  If it exist then a cc will be returned.

**Pull request merged**